rotkehlchen/exchanges/htx.py

```diff
@@ -0,0 +1,191 @@
+import base64
+import datetime
+import hashlib
+import hmac
+import logging
+import urllib.parse
+from typing import TYPE_CHECKING, Any, Final
+
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.converters import asset_from_htx
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_asset_amount
+from rotkehlchen.types import ApiKey, ApiSecret, Location
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import AssetWithOracles
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+HUOBI_BASE_URL: Final = 'https://api.huobi.pro'
+VALID_ACCOUNT_BALANCE_TYPES: Final = ('spot', 'otc', 'point')
+VALID_BALANCE_TYPES: Final = ('trade', 'frozen')
+
+
+class Htx(ExchangeInterface):
+    def __init__(
+            self,
+            name: str,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ):
+        super().__init__(
+            name=name,
+            location=Location.HTX,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+        self.valid_account_balances_types = VALID_ACCOUNT_BALANCE_TYPES
```

Comment:
you did not do what I asked here. Neither this nor valid_balance_types need to exist. Constant literals at the place they are used. Please read what I write.